src/main/java/com/amazon/opendistroforelasticsearch/ad/task/ADTaskManager.java

```diff
@@ -526,9 +603,11 @@ public void stopDetector(
             try (XContentParser parser = RestHandlerUtils.createXContentParserFromRegistry(xContentRegistry, searchHit.getSourceRef())) {
                 ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
                 ADTask adTask = ADTask.parse(parser, searchHit.getId());
+                logger.debug("latest task id is {}, for detector {}", adTask.getTaskId(), adTask.getDetectorId());
 
-                // TODO: support resetting realtime task as stopped
-                if (adTask.isHistoricalTask() && !isADTaskEnded(adTask) && lastUpdateTimeExpired(adTask)) {
+                // TODO: check realtime detector job and reset realtime task as stopped.
+                if (resetTaskState && adTask.isHistoricalTask() && !isADTaskEnded(adTask) && lastUpdateTimeExpired(adTask)) {
```

Comment:
when the coordinating node is gone, would the reset happen?

When the worker node is gone, would the coordinating node wait forever?

Reply:
Reply on new PR https://github.com/opensearch-project/anomaly-detection/pull/104/files#r660172804